source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,293 @@
+information about where {+odm+} persists the model:
+
+- ``client_name``: Retrieves the client name
+- ``database_name``: Retrieves the database name
+- ``collection_name``: Retrieves the collection name
+
+The following example shows how to retrieve and print the names of the client, 
+database, and collection where documents for the ``Band`` class are persisted:
+
+.. io-code-block::
+
+  .. input:: /includes/data-modeling/persistence-configuration.rb
+     :language: ruby
+     :start-after: start persistence context attributes
+     :end-before: end persistence context attributes
+
+  .. output::
+     :language: ruby
+     :visible: false
+
+     default
+
+     my_bands
+
+     bands
+
+Customize Your Persistence Configuration
+----------------------------------------
+
+{+odm+} provides both model-level and runtime options for customizing your 
+persistence configuration. The following sections describe these options.
+
+Model-Level Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Suppose you want to store your model's documents in a collection with a 
+different name than the pluralized form of the model class name.
+You can use the ``store_in`` macro to change the collection, database, or client
+where {+odm+} stores a model's documents. The following example shows how
+to use the ``store_in`` macro to store documents from the ``Person`` class in
+a collection called ``citizens`` in the ``other`` database within a client
+named ``analytics``:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in example
+   :end-before: end store_in example
+
+The ``store_in`` macro can also accept a lambda function. This is useful if you
+want to define a persistence context with values that cannot use a constant string. 
+
+You might want to use this pattern in a multi-tenant application,
+where multiple users share common access to an application. By using
+a lambda, you can define a persistence context based on the current thread so
+that users cannot access each others' data.
+
+The following example stores documents in a database determined by the 
+current thread:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in lambda example
+   :end-before: end store_in lambda example
+
+.. _mongoid-set-runtime-persistence-options:
+
+Runtime Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``with()`` method on a model class or instance to change
+a model's persistence configuration for a group of operations during runtime.
+
+Call the ``with()`` method on a model class or instance and pass options 
+that define a persistence context. The ``with()`` method accepts the following 
+arguments:
+
+- ``options_or_context``: A customizable set of options or a 
+  ``Mongoid::PersistenceContext`` object.
+
+- ``options``: A customizable set of options.
```

Comment:
These parameter names are a sad casualty of Ruby's lack of method overloading. What the net result, here, is that the `with` methods has (effectively) two different signatures:

- `with(context, options)` -- where `context` is an instance of `Mongoid::PersistanceContext`, and `options` is a Hash, and
- `with(options)` -- where all you supply is the options Hash.

I wonder if it might be clearer to document it this way, rather than exposing the `options_or_context` parameter, which is awkward to describe.